### PR TITLE
default env var for --param

### PR DESCRIPTION
**Pull request by gauron99**

### CODE REVIEW

File: http-server/main.go [hunk 1]
```diff
@@ -2,21 +2,65 @@ package main
...
+	}
+	return nil
+}
+
 func main() {
+	e := loadEnvSettings()
+	if e != nil {
+		log.Fatal(e) //print out error and exit
+	}
+	os.Setenv("SERVER_PARAM", "param but better") //custom env var
```
Comment (zroubalik) [inline http-server/main.go:53]: not needed, just set the ENV varialbe in the shell

File: http-server/main.go [hunk 2]
```diff
@@ -2,21 +2,65 @@ package main
...
 	var paramPtr string
 	flag.StringVar(&paramPtr, "param", "", "print this everywhere(long)")
 	flag.StringVar(&paramPtr, "p", "", "print this everywhere(short)")
-
-	// handle possible cli arguments & set
 	flag.Parse()
+
+	// have default parameter set as ENV variable if --param is not given
+	if paramPtr == "" {
+		paramPtr = os.Getenv("SERVER_PARAM")
```
Comment (zroubalik) [inline http-server/main.go:64]: Getenv retrieves the value of the environment variable named by the key. It returns the value, which will be empty if the variable is not present. To distinguish between an empty value and an unset value, use LookupEnv.

Comment (zroubalik) [reply http-server/main.go:64]: Let's use  `LookupEnv`, even though it doesn't make much sense, but just to try it out :)

File: http-server/main.go [hunk 3]
```diff
@@ -2,21 +2,65 @@ package main
...
+		line := scanner.Text()
+		if strings.HasPrefix(line, "#") {
+			continue //lines starting with '#' are comments
+		}
+		if !strings.Contains(line, "=") { //line has to contain '='
+			log.Println("Warning: env_file wasn't properly loaded; line in env file doesn't contain '=':", line)
+			return nil
+		}
+		split := strings.Split(line, "=")
+		os.Setenv(split[0], split[1])
```
Comment (zroubalik) [inline http-server/main.go:42]: It is probably not needed to set it back as ENV variable, but just pass the reference and directly work with the parsed values in the functions

**Comment by gauron99** [comment]
@zroubalik i did some fun stuff

File: http-server/server/server.go [hunk 4]
```diff
@@ -17,21 +18,59 @@ import (
 type serverInfo struct {
 	pageAccessCount int
 	param           string
+	variables       map[string]string
+}
+
+// LoadSettings reads data from a file called "env" in current dir
+// and sets all environment variables for this run.
+// Returns map of keys&values in the env file.
+func LoadSettings() (vars map[string]string, err error) {
```
Comment (zroubalik) [inline http-server/server/server.go:27]: You could move this to the constructor, ie `NewServerInfo()` and make this function private :)

**Review by zroubalik — COMMENTED** [review]
Almost there :)

File: http-server/server/server.go (hunk 4 above)
Comment (gauron99) [reply http-server/server/server.go]: good point! reworked now

File: http-server/server/server.go [hunk 5]
```diff
@@ -17,21 +18,63 @@ import (
...
 	}
 	return res
 }
 
 // MakeServerInfo creates & returns newly initialized serverInfo structure with given args
 func NewServerInfo(parameter string) *serverInfo {
-	return &serverInfo{0, parameter}
+	vars, e := loadSettings()
+	if e != nil {
+		log.Fatal(e) //print out error and exit
```
Comment (zroubalik) [inline http-server/server/server.go:75]: return error instead and exit in main

**Review by zroubalik — CHANGES_REQUESTED** [review]

**Review by zroubalik — APPROVED** [review]
LGTM

**Review by zroubalik — APPROVED** [review]

**Pull request merged**